GoogleForCreators/web-stories-wp
# Editor: Add collapsing carousel

## Context

This adds a collapsing carousel as per Figma designs.

## To-do


## User-facing changes

| State | Video |
|-|-|
| Wide page | ![carousel-collapse-wide-fast](https://user-images.githubusercontent.com/637548/142284566-dc417cc3-c29f-440f-8c32-cdba38ac607c.gif) |
| Narrow page with many thumbnails | ![carousel-collapse-narrow-multipage-fast](https://user-images.githubusercontent.com/637548/142284570-f8e2a62a-8143-4420-a5d9-0f96c7bf6760.gif) |
| Narrow page with a single thumbnail | ![carousel-collapse-narrow-singlepage-fast](https://user-images.githubusercontent.com/637548/142284607-d718e222-7d70-4d0b-a9bc-affd579bc87c.gif) |

## Testing Instructions

This PR can be tested by following these steps:

1. Observe there is a new bar above the page carousel with the current page number and the total page count
2. Click this bar to collapse the page carousel
3. Observe the collapse state with the page count bar perfectly aligning in the middle of the buttons to either side and that the canvas above has expanded to use the extra space
4. Click the bar again to expand the page carousel
5. Observe that everything works as expected and that the canvas has shrunk to make room for the carousel.





---

Fixes #9641
